mdp/hinet/layer.py

```diff
@@ -283,6 +314,23 @@ def __init__(self, nodes, dtype=None):
                 err = "The nodes have different input dimensions."
                 raise mdp.NodeException(err)
         output_dim = self._get_output_dim_from_nodes()
+
+        # store which nodes are pretrained up to what phase
+        _pretrained_phase = [node.get_current_train_phase()
+                             for node in nodes]
+        # check if all the nodes are already fully trained
+        train_len = 0
+        for i_node, node in enumerate(nodes):
+            if node.is_trainable():
+                train_len += (len(node._get_train_seq())
+                              - _pretrained_phase[i_node])
+        if train_len:
+            self._is_trainable = True
+            self._training = True
+        else:
+            self._is_trainable = False
+            self._training = False
```

Comment:
Sorry deleted my previous comment. The reason this part of the code is repeated is because SameInputLayer does not call its super where the flags are set. It instead calls Node's constructor.